### ISSUE TITLE
Compile Openddsharp with some fixes 

### ISSUE THREAD

**Issue by amielc1** [jmmorato/openddsharp#246]
I need to use DDS via RtpsRelay (where multicast is not allowed).
I discussed this with OpenDDS [about this issue](https://github.com/OpenDDS/OpenDDS/issues/4678) - they told me about the bug they resolved in the 3.28.1 version.
So, I need to compile OpenDDSSharp with some of the [**specific changes** ](https://github.com/OpenDDS/OpenDDS/pull/4485/commits/d77ef6fb53c3a72db00314d513082133a1fa5861)
Do you have a tutorial about how I can compile these changes? 

Thanks a lot, 
Amiel

**Comment by jmmorato** [comment]
Sorry for my late response but I have been in a business travel the last two weeks with no access to my personal computer...

No, there is no documentation yet about building OpenDDSharp but the github actions only requires to change the OpenDdsVersion parameter in order to rebuild with a new version:
https://github.com/jmmorato/openddsharp/blob/339a136a571b1d440783a4a5cd8f5fb3a1a57b4b/.github/workflows/ci_standard.yaml#L28
https://github.com/jmmorato/openddsharp/blob/339a136a571b1d440783a4a5cd8f5fb3a1a57b4b/.github/workflows/cd_standard.yaml#L17

However, I foresee breaking changes due to the new versions of OpenDDS that should be adapted for a successful compilation. 

I will work on that this week and add you as reviewer in the PR.

Thanks for your patience

**Comment by amielc1** [comment]
@jmmorato Thank you a lot. 
Please update me once you update OpenDDS version. 
Amiel

**Comment by amielc1** [comment]
@jmmorato Thank you for [your latest changes](https://www.nuget.org/packages/OpenDDSharp/3.281.1-beta20022), it's still  doesn't work - when I  use via **RTPS Relay** (in multicast everything ok )  -I asked this question in [OpenDDS ](https://github.com/OpenDDS/OpenDDS/issues/4678#issuecomment-2206338890)
Thank you very much for your special work!!!

**Comment by jmmorato** [comment]
Ok, please keep me informed and if you need something else I will try to implement ASAP.

**Comment by amielc1** [comment]
@jmmorato 
I want to make sure, 
I have this ini file for my Publisher : ( I took it from Open DDS Rtps Smoe Tests)  

>  [common]
> DCPSGlobalTransportConfig=$file
> DCPSThreadStatusInterval=1
> DCPSDebugLevel=10
> ORBDebugLevel=10
> ORBLogFile=Publisher.log
> 
> [domain/42]
> DiscoveryConfig=rtps_discovery
> 
> [rtps_discovery/rtps_discovery]
> SedpMulticast=0
> SpdpRtpsRelayAddress=127.0.0.1:4444
> SedpRtpsRelayAddress=127.0.0.1:4445
> RtpsRelayOnly=1
> SedpMaxMessageSize=1400
> 
> [transport/pub_rtps]
> transport_type=rtps_udp
> use_multicast=0
> DataRtpsRelayAddress=127.0.0.1:4446
> RtpsRelayOnly=1
> max_message_size=1400**

Do  I need to explicitly load "pub_rtps" transport config
Like use : 
`                TransportRegistry.Instance.BindConfig(configName, participant);
`
, or does Open DDS load it by default?

**Comment by jmmorato** [comment]
> Do I need to explicitly load "pub_rtps" transport config
No, it should be ok if you get the domain factory with the corresponding `-DCPSConfigFile` parameter.
